### PR TITLE
Feature/optimize path information

### PR DESCRIPTION
**Pull request by daniel-flemming**
This PR optimizes the PathInformation class for speed and memory usage.

### CODE REVIEW

File: src/Api.Rest.Dtos/PathHelper.cs [hunk 1]
```diff
@@ -153,139 +158,196 @@ public static PathInformationDto RoundtripString2PathInformation( [NotNull] stri
 				throw new ArgumentException( "The path string must not be null or empty.", nameof( path ) );
```
Comment (czthiele) [inline src/Api.Rest.Dtos/PathHelper.cs:158]: RoundtripString2PathInformation seems to be missing a check or usefull exception when the structure does not correspond with the actual path, e.g. "PP:part1/part2/part3". There was an exception before (The number of components in path string '{path}' and structure '{maybeStructure}' must be equal.)

Comment (czthiele) [reply src/Api.Rest.Dtos/PathHelper.cs:158]: Maybe you could also add some test for this to the PathHelperTests.

File: src/Api.Rest.Dtos/Api.Rest.Dtos.csproj [hunk 2]
```diff
@@ -41,8 +41,15 @@
...
-  <ItemGroup>
+    <ItemGroup Label=".NET references" Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
+      <PackageReference Include="Microsoft.Bcl.HashCode" Version="1.1.1" />
+      <PackageReference Include="System.Buffers" Version="4.5.1" />
+      <PackageReference Include="System.Memory" Version="4.5.5" />
+    </ItemGroup>
+
+    <ItemGroup>
     <PackageReference Include="JetBrains.Annotations" Version="2020.3.0" />
+    <PackageReference Include="Microsoft.Extensions.ObjectPool" Version="6.0.6" />
```
[the remark this answers is not in the document]
Comment (daniel-flemming) [reply src/Api.Rest.Dtos/Api.Rest.Dtos.csproj:52]: Done. Thanks.

File: src/Api.Rest.Dtos/Data/PathInformationDto.cs [hunk 3]
```diff
@@ -309,18 +316,17 @@ public override bool Equals( object obj )
...
 		public override int GetHashCode()
 		{
-			if( !_HashCode.HasValue )
-			{
-				var hash = 1;
-				foreach( var elem in this )
-				{
-					hash ^= elem.GetHashCode();
-				}
+			if( _HashCode != -1 )
```
[the remark this answers is not in the document]
Comment (daniel-flemming) [reply src/Api.Rest.Dtos/Data/PathInformationDto.cs:319]: Yes. Thanks for spotting this bug. Fixed it.

File: src/Api.Rest.Dtos/Data/PathInformationDto.cs [hunk 4]
```diff
@@ -167,7 +167,14 @@ public bool IsBelow( PathInformationDto path )
 			if( path == null || path.Count > Count )
 				return false;
 
-			return !path.Where( ( t, i ) => !this[ i ].Equals( t ) ).Any();
+			var i = 0;
```
[the remark this answers is not in the document]
Comment (daniel-flemming) [reply src/Api.Rest.Dtos/Data/PathInformationDto.cs:170]: Done.

File: src/Api.Rest.Dtos/PathHelper.cs (hunk 1 above)
Comment (daniel-flemming) [reply src/Api.Rest.Dtos/PathHelper.cs:158]: Fixed. Thanks for the perfect help by writing tests.

File: src/Api.Rest.Dtos/Api.Rest.Dtos.csproj [hunk 5]
```diff
@@ -41,8 +41,15 @@
     <SupportedPlatform Include="browser" />
   </ItemGroup>
 
-  <ItemGroup>
+    <ItemGroup Label=".NET references" Condition="'$(TargetFrameworkIdentifier)' != '.NETCoreApp'">
```
[the remark this answers is not in the document]
Comment (daniel-flemming) [reply src/Api.Rest.Dtos/Api.Rest.Dtos.csproj:44]: I think so. .Net Core will bring a few classes (HasCode, Span etc.) that need to be referenced as Nugget packages for .Net Standard / .Net Framework.

**Pull request merged**